plugins/modules/nd_setup.py

```diff
@@ -0,0 +1,471 @@
+    proxy_password = nd.params.get("proxy_password")
+    ignore_proxy = nd.params.get("ignore_proxy")
+    dns_search_domain = nd.params.get("dns_search_domain")
+    app_network = nd.params.get("app_network")
+    service_network = nd.params.get("service_network")
+    app_network_ipv6 = nd.params.get("app_network_ipv6")
+    service_network_ipv6 = nd.params.get("service_network_ipv6")
+    ntp_config = nd.params.get("ntp_config")
+    nodes = nd.params.get("nodes")
+    state = nd.params.get("state")
+
+    if state == "query":
+        nd.existing = nd.request("/clusterstatus/install", method="GET")
+    else:
+        if len(cluster_name) > 63:
+            nd.fail_json("A length of 1 to 63 characters is allowed.")
+        elif len(re.findall(r"[^a-zA-Z0-9-]", cluster_name)) > 0:
+            nd.fail_json("Valid characters include letters, digits and hyphen.")
+        elif len(re.findall(r"^-|-$", cluster_name)) > 0:
+            nd.fail_json("The name cannot start or end with a hyphen.")
+
+        payload = {
+            "clusterConfig": {
+                "name": cluster_name,
+                "ntpServers": ntp_server,
+                "nameServers": dns_server,
+                "ntpConfig": {
+                    "servers": [
+                        {
+                            "host": server.get("ntp_host"),
+                            "keyID": server.get("ntp_key_id"),
+                            "prefer": server.get("preferred"),
+                        }
+                        for server in ([] if ntp_config is None else ntp_config.get("servers"))
+                    ],
+                    "keys": [
+                        {
+                            "id": key.get("ntp_key_id"),
+                            "key": key.get("ntp_key"),
+                            "authType": key.get("authentication_type"),
+                            "trusted": key.get("trusted"),
+                        }
+                        for key in ([] if ntp_config is None else ntp_config.get("keys", []))
+                    ],
+                },
+                "searchDomains": dns_search_domain,
+                "ignoreHosts": ignore_proxy,
+                "proxyServers": [
+                    {
+                        "proxyURL": proxy_server,
+                        "username": proxy_username,
+                        "password": proxy_password,
+                    }
+                ],
+                "appNetwork": app_network,
+                "serviceNetwork": service_network,
+                "appNetworkV6": app_network_ipv6,
+                "serviceNetworkV6": service_network_ipv6,
+            },
+            "nodes": [
+                {
+                    "hostName": node.get("hostname"),
+                    "serialNumber": node.get("serial_number"),
+                    "dataNetwork": {
+                        "ipSubnet": node.get("data_network").get("ipv4_address"),
```

Comment:
If using .get() because "data_network" can not be present. Then you want to return an object that can be passed to the next get and not fail because .get() is not supported on None type.
```suggestion
                        "ipSubnet": node.get("data_network", {}).get("ipv4_address"),
```
This applies to all chained .get() in this file